qutebrowser/misc/sql.py

```diff
@@ -425,35 +433,68 @@ def delete(self, field, value):
             raise KeyError('No row with {} = "{}"'.format(field, value))
         self.changed.emit()
 
-    def _insert_query(self, values, replace):
+    def _insert_query(self, values, replace=False, ignore=False):
         params = ', '.join(':{}'.format(key) for key in values)
-        verb = "REPLACE" if replace else "INSERT"
+        verb = 'INSERT'
+        if replace:
+            verb += ' OR REPLACE'
+        elif ignore:
+            verb += ' OR IGNORE'
         return Query("{verb} INTO {table} ({columns}) values({params})".format(
             verb=verb, table=self._name, columns=', '.join(values),
             params=params))
 
-    def insert(self, values, replace=False):
+    def insert(self, values, replace=False, ignore=False):
         """Append a row to the table.
 
         Args:
             values: A dict with a value to insert for each field name.
             replace: If set, replace existing values.
+            ignore: If set, ignore conflicts.
         """
-        q = self._insert_query(values, replace)
-        q.run(**values)
+        q = self._insert_query(values, replace, ignore)
+        result = q.run(**values)
         self.changed.emit()
+        return result
 
-    def insert_batch(self, values, replace=False):
+    def insert_batch(self, values, replace=False, ignore=False):
         """Performantly append multiple rows to the table.
 
         Args:
             values: A dict with a list of values to insert for each field name.
             replace: If true, overwrite rows with a primary key match.
+            ignore: If set, ignore conflicts.
         """
-        q = self._insert_query(values, replace)
+        q = self._insert_query(values, replace, ignore)
         q.run_batch(values)
         self.changed.emit()
 
+    def update(self, update, where, escape=True):
+        """Execute update rows statement.
+
+        Args:
+            update: column:value dict with new values to set
+            where: column:value dict for filtering
+            escape: enable new values SQL-escaping
+        """
+        if escape:
+            u = ', '.join(('{k} = :{k}'.format(k=k) for k in update.keys()))
```

Comment:
Did we have this in the past for `insert`? I don't remember exactly why I added this or where I got it from. If we don't need it at all and we don't have this same option for `insert` then I guess it doesn't make sense to keep it.

Reply:
I remembered, I got this from #4403. I'll evaluate if we should keep this, as its not fully tested either.

Reply:
I ended up removing this, but it is possible that we could need something similar in the future. Although I would like an approach in which I can pass some literal values that need escaping and some other values that don't, instead of being just one or the other.